common-messaging/src/main/scala/org/genivi/sota/messaging/kafka/MessageListener.scala

```diff
@@ -15,33 +15,44 @@ object MessageListener {
 
   type MsgParser[T <: BusMessage] = T => Future[_]
 
-  def buildSource[T <: BusMessage](fromSource: Source[CommittableMessage[Array[Byte], T], NotUsed],
+  /**
+    * This trait is used to enable NATS to pretend it can do committable messages
+    */
+  trait CommittableMessageLike[T <: BusMessage] {
+    def getMsg(): T
+    def commitRecord(): Future[Done]
+  }
+
+  class NatsMsg[T <: BusMessage](msg: T) extends CommittableMessageLike[T] {
+    override def getMsg(): T = msg
+
+    override def commitRecord(): Future[Done] = Future.successful(Done)
+  }
+
+  class KafkaMsg[T <: BusMessage](msg: CommittableMessage[_, T]) extends CommittableMessageLike[T] {
+    override def getMsg(): T = msg.record.value()
+
+    override def commitRecord(): Future[Done] = msg.committableOffset.commitScaladsl()
```

Comment:
This could probably return `Future[T]` instead, and this would simplify the pipeline as you  wouldn't need to `map(_ => msg)` (but you'd probably need to pass an implicit ec here)